src/qz/installer/TaskControl.java

```diff
@@ -0,0 +1,180 @@
+            killCmd = KILL_PID_CMD_WIN32;
+        } else {
+            javaProcs = ShellUtilities.executeRaw(JAVA_PID_QUERY_POSIX).split("\\s*\\r?\\n");
+            trayProcs = ShellUtilities.executeRaw(TRAY_PID_QUERY_POSIX).split("\\s*\\r?\\n");
+            selfProc = MacUtilities.getProcessID(); // Works for Linux too
+            killCmd = KILL_PID_CMD_POSIX;
+        }
+        if (javaProcs.length > 0) {
+            // Find intersections of java and qz-tray.jar
+            List<String> intersections = new ArrayList<>(Arrays.asList(trayProcs));
+            intersections.retainAll(Arrays.asList(javaProcs));
+
+            // Remove any instances created by this installer
+            intersections.remove("" + selfProc);
+
+            // Kill whatever's left
+            for (String pid : intersections) {
+                // isNumeric() needed for Windows; filters whitespace, headers
+                if(StringUtils.isNumeric(pid)) {
+                    // Set last command to the pid
+                    killCmd[killCmd.length -1] = pid;
+                    success = success && ShellUtilities.execute(killCmd);
+                }
+            }
+        }
+
+        // Use jcmd to kill class processes too, such as through the IDE
+        if(SystemUtilities.isJDK()) {
+            String[] procs = ShellUtilities.executeRaw("jcmd", "-l").split("\\r?\\n");
+            for(String proc : procs) {
+                String[] parts = proc.split(" ", 1);
+                if (parts.length >= 2 && parts[1].contains(PrintSocketServer.class.getCanonicalName())) {
+                    killCmd[killCmd.length - 1] = parts[0].trim();
+                    success = success && ShellUtilities.execute(killCmd);
+                }
+            }
+        }
+
+        if(SystemUtilities.isWindowsXP()) {
+            File f = new File("TempWmicBatchFile.bat");
+            if(f.exists()) {
+                f.deleteOnExit();
+            }
+        }
+
+        return success;
+    }
+
+    public static ArrayList<Path> locateProcessPaths(boolean exactMatch, String ... ProcessNames) throws IOException {
+        ArrayList<Path> pathList = new ArrayList<>();
+        ArrayList<String> pidArray = getRootPIDs(exactMatch, ProcessNames);
+        if (SystemUtilities.isWindows()) {
+            for(String pid : pidArray) {
+                WIN32_PATH_QUERY[WIN32_PATH_QUERY_INPUT_INDEX] = "ProcessId=" + pid;
+                String[] response = ShellUtilities.executeRaw(WIN32_PATH_QUERY).split("\\s*\\r?\\n");
+                if (response.length > 1) {
+                    pathList.add(Paths.get(response[1]).toRealPath());
+                }
+            }
+        } else {
+            for(String pid : pidArray) {
+                pathList.add(Paths.get("/proc/", pid, "/exe").toRealPath());
+            }
+        }
+        return pathList;
+    }
+
+    public static ArrayList<String> getRootPIDs(boolean exactMatch, String ... ProcessNames) {
```

Comment:
I spy an uppercase variable name.

Reply:
Aso, this appears to be a Windows-only function.  Avoiding JNA contamination is important, we might need to split the `TaskControl` into a per platform class like we do with Installer, etc.

Reply:
This has been refactored into Windows-specific classes.  Marking as resolved.